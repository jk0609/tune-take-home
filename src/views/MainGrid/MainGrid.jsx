import React, { useState } from 'react';
// Data and helper functions
import logs from '../../data/logs';
import users from '../../data/users';
import { parseLogs, sortByProp } from '../../helpers/mainGridHelpers';

// Components
import GridCard from './components/GridCard';
import StyledMainGridWrapper from './styledComponents/StyledMainGridWrapper';

const userData = parseLogs(users, logs);

const MainGrid = () => {
  let [sortProp, changeSortProp] = useState('name');

  const generateUserCards = () => {
    // Calls sort helper, passing in name of prop to sort by
    let sortedUserData = sortByProp(sortProp, userData);

    // Generates a GridCard component for each user, passing down user props
    return sortedUserData.map(user => <GridCard key={user.id} {...user} />);
  };

  return (
    <StyledMainGridWrapper>
      <label htmlFor="sort-dropdown">
        Sort results by:
        <select
          id="sort-dropdown"
          onChange={e => changeSortProp(e.target.value)}
        >
          <option value="name">Name</option>
          <option value="impressions">Impressions</option>
          <option value="conversions">Conversions</option>
          <option value="revenue">Revenue</option>
        </select>
      </label>
      <div className="main-grid">{generateUserCards()}</div>
    </StyledMainGridWrapper>
  );
};
export default MainGrid;
