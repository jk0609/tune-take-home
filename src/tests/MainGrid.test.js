import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainGrid from '../views/MainGrid/MainGrid';

configure({ adapter: new Adapter() });

jest.mock('../data/users', () => [
  {
    name: 'Charlie S. Gerardi',
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/brad_frost/128.jpg',
    id: 0,
    occupation: 'Residential electrician'
  }
]);

jest.mock('../data/logs', () => [
  {
    time: '2013-04-19 15:26:10',
    type: 'conversion',
    user_id: 0,
    revenue: 10
  },
  {
    time: '2013-04-24 19:20:09',
    type: 'impression',
    user_id: 0,
    revenue: 0
  },
  {
    time: '2013-04-16 07:20:32',
    type: 'impression',
    user_id: 0,
    revenue: 0
  }
]);

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MainGrid />);
});

describe('MainGrid snapshot', () => {
  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
