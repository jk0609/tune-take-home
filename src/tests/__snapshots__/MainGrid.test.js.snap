// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainGrid snapshot should match snapshot 1`] = `
<styled.div>
  <label
    htmlFor="sort-dropdown"
  >
    Sort results by:
    <select
      id="sort-dropdown"
      onChange={[Function]}
    >
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="impressions"
      >
        Impressions
      </option>
      <option
        value="conversions"
      >
        Conversions
      </option>
      <option
        value="revenue"
      >
        Revenue
      </option>
    </select>
  </label>
  <div
    className="main-grid"
  >
    <GridCard
      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/brad_frost/128.jpg"
      conversions={
        Object {
          "count": 1,
          "graphData": Object {
            "04/19": 10,
          },
        }
      }
      id={0}
      impressions={2}
      key="0"
      name="Charlie S. Gerardi"
      occupation="Residential electrician"
      revenue={10}
    />
  </div>
</styled.div>
`;
